server/services/version.ts

```diff
@@ -93,6 +93,14 @@ export async function createVersion(user: UserDoc, data: CreateVersion) {
     throw Forbidden({ data }, 'Unable to create version, failed permissions check.')
   }
 
+  let allVersionsForModel = VersionModel.find({ model: version.model })
+  if (
+    (await allVersionsForModel).filter((versionDuplicateCheck) => versionDuplicateCheck.version === version.version)
```

Comment:
You shouldn't be doing this logic on the server, you should be getting Mongo to only find models with the specific model _and_ the specified version.

Reply:
Resolved